jenkinsci/plasticscm-plugin
# Add Mergebot features

This PR brings the functionality currently located in the [Mergebot plugin](https://github.com/jenkinsci/plasticscm-mergebot-plugin).

Installing the plugin will enable DevOps workflows from Plastic SCM to be triggered in Jenkins. Mergebots trigger builds, setting the spec to build in a specific parameter. The Mergebot SCM picks that up and uses it to checkout the sources.

The benefit of this merge is keeping a single codebase for both SCM plugins, and unlock the latest features in the Mergebot plugin: `cm` as auto-installing tool, setting per-project credentials, cleanup modes.

This PR is still a draft. Pending to do:
* Manual testing
* Update documentation

### Testing done

There are a few unit tests added, and the functionality was verified via manual testing.

Tests include:
* Verify that the Plastic SCM VCS plugin is still working as before
    * Freestyle projects
    * Pipeline projects with hard-coded script in project config
    * Pipeline projects from SCM
* Verify that the new Mergebot VCS plugin works as expected
    * Freestyle projects
    * Pipeline projects with hard-coded script in project config
    * Pipeline projects from SCM
* Verify that installing the plugin in an existing environment doesn't break the configuration


### Submitter checklist
